internal/rbd/rbd_util.go

```diff
@@ -578,36 +600,34 @@ func (rv *rbdVolume) getCloneDepth(ctx context.Context) (uint, error) {
 	vol.Monitors = rv.Monitors
 	vol.RbdImageName = rv.RbdImageName
 	vol.conn = rv.conn.Copy()
-
+	currVol := vol
+	defer currVol.Destroy()
 	for {
-		if vol.RbdImageName == "" {
+		if currVol.RbdImageName == "" {
 			return depth, nil
 		}
-		err := vol.openIoctx()
+		err := currVol.openIoctx()
 		if err != nil {
 			return depth, err
 		}
-
-		err = vol.getImageInfo()
-		// FIXME: create and destroy the vol inside the loop.
-		// see https://github.com/ceph/ceph-csi/pull/1838#discussion_r598530807
-		vol.ioctx.Destroy()
-		vol.ioctx = nil
+		parentVol, err := currVol.getImageParent()
+		defer parentVol.Destroy()
+		currVol.ioctx.Destroy()
+		currVol.ioctx = nil
```

Comment:
why this is required here?

Reply:
Don't destroy the iocontext here, it should automatically be done when the rbdVolume is not needed anymore. If there is a leak, the `defer currVol.Destroy()` needs to be placed at the tight spot (somewhere inside the for loop).